zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZookeeperRegisterBuilder.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.zookeeper;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.zookeeper.NodeValueCodec;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerListener;
+import com.linecorp.armeria.server.ServerListenerBuilder;
+
+/**
+ * Builds a new {@link ServerListener}, which registers the server to the zookeeper instance.
+ * <h2>Example</h2>
+ * <pre>{@code
+ * ServerListener register = new ZookeeperRegisterBuilder()
+ * // Zookeeper Connection String
+ * .connect("myZooKeeperHost:2181")
+ * // Zookeeper Node to use
+ * .node("/myProductionEndpoints")
+ * // Session Timeout
+ * .sessionTimeOut(10000)
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ * */
+public class ZookeeperRegisterBuilder {
```

Comment:
- `final`?
- I'm curious what other options you have plan to add. Could you answer?


After thinking a little bit, I'm somewhat concerned about what this PR really tries to achieve. If you are interested in adding a builder for `ZooKeeperUpdatingListener`, how about:

- renaming to `ZooKeeperUpdatingListenerBuilder`
- making `build()` return `ZooKeeperUpdatingListener`
- replacing the old `ZooKeeperUpdatingListener` with what you built in `build()`, i.e. making `ZooKeeperUpdatingListener` use `ServerListenerBuilder` internally (or you can just stop using `ServerListenerBuilder` in this case because it's so simple.)
- continuing exposing the static factory methods in `ZooKeeperUpdatingListener` so that a user doesn't need to use the builder API for simple use cases

This way, users wouldn't worry much about backward incompatibility. 

If you want to do some evolutionary API changes to introduce Curator dependency, you don't really worry about breaking the API - just rewrite `ZooKeeperUpdatingListener`.